### PR TITLE
show older scan job report

### PR DESCRIPTION
**Pull request by shaik80**
Signed-off-by: shaik80 <sh.mudassir98@gmail.com>

### :nut_and_bolt: Description: What code changed, and why?

In this PR, there is a fix to show old scan report, previously it use to latest scan report because of this old scan report was shown empty
### :chains: Related Resources
https://chefio.atlassian.net/browse/SHIELD-54

### :+1: Definition of Done
You need to see old san report after click on report as shown in below video

### :athletic_shoe: How to Build and Test the Change
```
➜  automate git:(SHIELD-54/show_older_scan_job_report_fix) t_fix) hab studio enter
[158][default:/src:130]# start_all_services
......
......
......
[158][default:/src:130]# rebuild components/compliance-service/
[158][default:/src:130]# rebuild components/automate-deployment/

run 2 scan jobs an try to see the data
```

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable

https://user-images.githubusercontent.com/47217471/172142259-b8d51fe2-270a-49e7-ae25-4a6fc506d4ef.mov

### CODE REVIEW

File: components/compliance-service/reporting/relaxting/stats.go [hunk 1]
```diff
@@ -20,7 +20,13 @@ func (backend ES2Backend) GetStatsSummary(filters map[string][]string) (*stats.R
...
 	// Only end_time matters for this call
 	filters["start_time"] = []string{}
-	depth, err := backend.NewDepth(filters, true)
+	latestOnly := true
+
+	if filters["job_id"] != nil {
+		latestOnly = false
+	}
+
+	depth, err := backend.NewDepth(filters, latestOnly)
```
[the remark this answers is not in the document]
Comment (shaik80) [reply components/compliance-service/reporting/relaxting/stats.go:29]: yes we can

File: components/compliance-service/reporting/relaxting/stats.go [hunk 2]
```diff
@@ -168,7 +175,12 @@ func (backend ES2Backend) GetStatsFailures(reportTypes []string, size int, filte
 
 	// Only end_time matters for this call
 	filters["start_time"] = []string{}
-	depth, err := backend.NewDepth(filters, true)
+
+	latestOnly := FetchLatestDataOrNot(filters)
+
+	depth, err := backend.NewDepth(filters, latestOnly)
+	logrus.Println(latestOnly, filters["job_id"], "scanIssue reportType")
```
Comment (atultherajput) [inline components/compliance-service/reporting/relaxting/stats.go:182]: Do we need to log this message? If yes, then can we change the level to debug?

File: components/compliance-service/reporting/relaxting/util.go [hunk 3]
```diff
@@ -291,3 +291,14 @@ func StringTagsFromProtoFields(tKey string, tValue *structpb.Value) *ESInSpecRep
 	}
 	return nil
 }
+
+func FetchLatestDataOrNot(filters map[string][]string) bool {
+	fmt.Println(filters)
```
Comment (atultherajput) [inline components/compliance-service/reporting/relaxting/util.go:296]: Is this print statement required? If yes, then can we replace it with a log message?

File: components/compliance-service/reporting/relaxting/util.go [hunk 4]
```diff
@@ -291,3 +291,14 @@ func StringTagsFromProtoFields(tKey string, tValue *structpb.Value) *ESInSpecRep
...
+	latestOnly := true
+	if filters["job_id"] != nil {
+		fmt.Println(filters, 222)
```
Comment (atultherajput) [inline components/compliance-service/reporting/relaxting/util.go:299]: Same comment as above.

**Review by atultherajput — APPROVED** [review]

**Review by vivekshankar1 — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/59958706/172569478-685abfb1-4006-4c7c-a900-014ca01d1b6c.mov

Last scan job 3: data was present 
but older scan job data like profile, nodes are not visible

**Review by vivekshankar1 — APPROVED** [review]
Works fine, thanks

**Review by vivek-yadav — APPROVED** [review]

**Pull request merged**